### PR TITLE
feat: support per collection public_folder

### PR DESCRIPTION
**Pull request by erezrokah**
Fixes https://github.com/netlify/netlify-cms/issues/3053

@erquhart let me know if this is what you had in mind (I'm not sure I completely understand the scenario in the issue).

We can move forward on the other related issues after this is merged.

### CODE REVIEW

File: packages/netlify-cms-core/src/actions/__tests__/config.spec.js [hunk 1]
```diff
@@ -101,5 +101,55 @@ describe('config', () => {
         ]),
       ).toEqual('_');
     });
+
+    it('should default public_folder to media_folder', () => {
```
Comment (erquhart) [inline packages/netlify-cms-core/src/actions/__tests__/config.spec.js]: We should split the collection specific public_folder/media_folder testing to a separate describe block for clarity (same description as top level test).

File: packages/netlify-cms-core/src/actions/__tests__/config.spec.js [hunk 2]
```diff
@@ -101,5 +101,55 @@ describe('config', () => {
...
+      expect(
+        applyDefaults(
+          fromJS({
+            collections: [{ folder: 'foo', path: '{{slug}}/index' }],
+          }),
+        ).get('collections'),
+      ).toEqual(
+        fromJS([{ folder: 'foo', path: '{{slug}}/index', media_folder: '', public_folder: '' }]),
+      );
+    });
```
Comment (erquhart) [inline packages/netlify-cms-core/src/actions/__tests__/config.spec.js]: We should add some tests to ensure collection level media_folder and public_folder default to the global values when not set.

Comment (erezrokah) [reply packages/netlify-cms-core/src/actions/__tests__/config.spec.js]: That logic is actually part of the selectors:
https://github.com/netlify/netlify-cms/blob/476f45096efa1723936a73f2e2e04d5c7ccd293f/packages/netlify-cms-core/src/reducers/entries.ts#L148
https://github.com/netlify/netlify-cms/blob/476f45096efa1723936a73f2e2e04d5c7ccd293f/packages/netlify-cms-core/src/reducers/entries.ts#L193

And tested here:
https://github.com/netlify/netlify-cms/blob/476f45096efa1723936a73f2e2e04d5c7ccd293f/packages/netlify-cms-core/src/reducers/__tests__/entries.spec.js#L80
and here:
https://github.com/netlify/netlify-cms/blob/476f45096efa1723936a73f2e2e04d5c7ccd293f/packages/netlify-cms-core/src/reducers/__tests__/entries.spec.js#L185

Collection based `media_folder` folders values are relative to the entry path so they have different semantics than the global ones

File: packages/netlify-cms-core/src/actions/__tests__/config.spec.js (hunk 1 above)
Comment (erezrokah) [reply packages/netlify-cms-core/src/actions/__tests__/config.spec.js]: Created a bit of nesting, but still better from before. Also aligned all the test titles to the same format

File: packages/netlify-cms-core/src/actions/__tests__/config.spec.js (hunk 2 above)
Comment (erquhart) [reply packages/netlify-cms-core/src/actions/__tests__/config.spec.js]: Ah, thanks for the pointers, sounds good 👍

**Pull request merged**